### PR TITLE
test: supercharged twap e2e

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #4844 

## What is the purpose of the change

Extends E2E test to check if CL twap properly
- errors when first created
- errors when all liquidity is removed
- updates as positions are created and swaps happen

## Testing and Verifying

E2E test suite directly updated

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: x/concentrated-liquidity/incentives.go [hunk 1]
```diff
@@ -151,6 +151,11 @@ func (k Keeper) prepareBalancerPoolAsFullRange(ctx sdk.Context, clPoolId uint64,
 	}
 	bondedShares := k.lockupKeeper.GetLockedDenom(ctx, gammtypes.GetPoolShareDenom(canonicalBalancerPoolId), longestDuration)
 
+	// We fail quietly if the Balancer pool has no bonded shares.
+	if bondedShares.IsZero() {
+		return 0, sdk.ZeroDec(), nil
+	}
+
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/incentives.go:158]: This is a drive-by change I found when I was attempting to check the spot price for the CL pool when it was connected to a Balancer pool that had no bonded shares. This is super unlikely to happen in prod, but should fail quietly here if we do somehow reach this logic branch. Please lmk what you all think (dont want this to just sneak in)

File: tests/e2e/e2e_test.go [hunk 2]
```diff
@@ -1610,6 +1679,11 @@ func (s *IntegrationTestSuite) TestAConcentratedLiquidity_CanonicalPool_And_Para
 
 	s.Require().True(found, "concentrated liquidity pool denom not found in superfluid assets")
 
+	// Check that the community pool module account possesses a position
+	communityPoolAddress := chainANode.QueryCommunityPoolModuleAccount()
+	positions := chainANode.QueryConcentratedPositions(communityPoolAddress)
+	s.Require().Len(positions, 1)
+
```
Comment (czarcas7ic) [inline tests/e2e/e2e_test.go:1686]: Also a drive by change, just checking that after the upgrade, the community pool in fact owns one position

File: tests/e2e/e2e_test.go [hunk 3]
```diff
@@ -922,6 +923,74 @@ func (s *IntegrationTestSuite) TestCreateConcentratedLiquidityPoolVoting() {
...
+	s.Require().Equal(sdk.Dec{}, afterRemoveTwapBOverA)
+
+	// Create a position and check that TWAP now returns a value.
+	s.T().Log("creating position")
+	chainANode.CreateConcentratedPosition(address1, "[-120000]", "40000", fmt.Sprintf("10000000%s,10000000%s", concentratedPool.GetToken0(), concentratedPool.GetToken1()), 0, 0, concentratedPool.GetId())
+	chainA.WaitForNumHeights(1)
+	timeAfterSwapRemoveAndCreatePlus1Height := chainANode.QueryLatestBlockTime()
+	secondTwapBOverA, err := chainANode.QueryGeometricTwapToNow(concentratedPool.GetId(), concentratedPool.GetToken0(), concentratedPool.GetToken1(), timeAfterSwapRemoveAndCreatePlus1Height)
+	s.Require().NoError(err)
+	s.Require().Equal(sdk.NewDec(1), secondTwapBOverA)
```
Comment (AlpinYukseloglu) [inline tests/e2e/e2e_test.go:993]: Can we clarify in comments where this `sdk.NewDec(1)` came from?

File: tests/e2e/e2e_test.go [hunk 4]
```diff
@@ -891,7 +891,7 @@ func (s *IntegrationTestSuite) TestSuperfluidVoting() {
 	)
 }
 
-func (s *IntegrationTestSuite) TestCreateConcentratedLiquidityPoolVoting() {
+func (s *IntegrationTestSuite) TestCreateConcentratedLiquidityPoolVoting_And_TWAP() {
```
Comment (AlpinYukseloglu) [inline tests/e2e/e2e_test.go:894]: Out of curiosity, why is this change paired with CL voting tests?

File: tests/e2e/configurer/chain/queries.go [hunk 5]
```diff
@@ -642,3 +642,24 @@ func (n *NodeConfig) QueryAllSuperfluidAssets() []superfluidtypes.SuperfluidAsse
...
+	out, _, err := n.containerManager.ExecCmd(n.t, n.Name, cmd, "")
+	require.NoError(n.t, err)
+	var result map[string][]interface{}
+	err = json.Unmarshal(out.Bytes(), &result)
+	require.NoError(n.t, err)
+	for _, acc := range result["accounts"] {
+		account, ok := acc.(map[string]interface{})
+		require.True(n.t, ok)
+		if account["name"] == "distribution" {
+			moduleAccount, ok := account["base_account"].(map[string]interface{})["address"].(string)
```
Comment (AlpinYukseloglu) [inline tests/e2e/configurer/chain/queries.go:658]: Screenshot 2023-05-29 at 5 19 54 PM

**Review by AlpinYukseloglu — APPROVED** [review]

File: tests/e2e/e2e_test.go (hunk 4 above)
Comment (czarcas7ic) [reply tests/e2e/e2e_test.go:894]: The next thing we need to work on is parallelizing e2e, so pairing some logic together that doesn't require one to be run before the other makes sense imo

**Review by p0mvn — APPROVED** [review]
